packages/react/src/primitives/SearchField/SearchField.tsx

```diff
@@ -0,0 +1,101 @@
+import classNames from 'classnames';
+import * as React from 'react';
+
+import { ComponentClassNames } from '../shared/constants';
+import { TextField } from '../TextField';
+import { FieldClearButton } from '../Field';
+import { SearchFieldButton } from './SearchFieldButton';
+import { strHasLength } from '../shared/utils';
+import { SearchFieldProps, InputProps } from '../types';
+
+const ESCAPE_KEY = 'Escape';
+const ENTER_KEY = 'Enter';
+const DEFAULT_KEYS = [ESCAPE_KEY, ENTER_KEY];
+
+export const useSearchField = (onSubmit: SearchFieldProps['onSubmit']) => {
+  const [value, setValue] = React.useState<string>('');
+
+  const clearValue = React.useCallback(() => setValue(''), [setValue]);
+
+  const onKeyDown: InputProps['onKeyDown'] = React.useCallback((event) => {
+    const key = event.key;
+
+    if (DEFAULT_KEYS.includes(key)) {
+      event.preventDefault();
+    }
+    if (key === ESCAPE_KEY) {
+      clearValue();
+    }
+    if (key === ENTER_KEY) {
+      onSubmitHandler(event.currentTarget.value);
```

Comment:
I see. That makes sense then. But I think in this case, it might be worth it for consistency. When the value changes, its child component(`TextField`) has to be re-rendered anyway. We don't actually get too much benefit from this unchanged prop(`onKeyDown`). It is like just saving a variable kind of thing.

Reply:
Well, recreating a function does have a memory cost. However, with `React.useCallback` and `[value]` dependency, we'll only recreate it when value changes, not on unrelated renders. I'll update to use value.